tests/gradients/test_spsa_gradient.py

```diff
@@ -443,10 +463,11 @@ def reference_qnode(x):
         assert np.isclose(qnode(par).item().val, reference_qnode(par))
         assert np.isclose(qml.jacobian(qnode)(par).item().val, qml.jacobian(reference_qnode)(par))
 
-    @pytest.mark.parametrize("num_directions, tol", [(100, 0.1), (500, 5e-3)])
+    @pytest.mark.parametrize("num_directions, tol", [(1, 0.1), (1, 5e-3)])
```

Comment:
Why this change?

Reply:
Ah, sorry. I tested whether this test really tests what it says. And it doesn't: For a single parameter, there is no stochasticity in SPSA. Will modify so that this test does what is says.

Reply:
now testing two parameters, so that there is a convergence to test for.
Now requires 10 seconds on my laptop for the many-shots test, do we want to keep it?

Reply:
Sorry, which is the "many-shots" test?

Reply:
Ah, oops. many directions! 
The second parametrized test here (which now uses 10k directions)

Reply:
Ah! I see... could we reduce the number of dimensions? Or just test it with 100 dimensions and remove the second parametrization

Reply:
I increased the tolernaces a bit and this allowed to do 100 and 1000 directions, which both is fast enough :)